File: src/main/scala/net/machinemuse/powersuits/client/render/modelspec/ModelPartSpec.scala
//package net.machinemuse.powersuits.client.render.modelspec
//
//import net.minecraft.nbt.NBTTagCompound
//
///**
//  * Ported to Java by lehjr on 11/8/16.
//  */
//class ModelPartSpec(val modelSpec: ModelSpec,
//                    val morph: MorphTarget,
//                    val partName: String,
//                    val slot: Int,
//                    val defaultcolourindex: Int = 0,
//                    val defaultglow: Boolean = false,
//                    val displayName: String) {
//
//  def getTexture(nbt: NBTTagCompound): String = {
//    if (nbt hasKey "texture") nbt getString "texture" else modelSpec.textures.head
//  }
//
//  def setTexture(nbt: NBTTagCompound, s: String) {
//    if (s.equals("") || s.equalsIgnoreCase(modelSpec.textures.head)) nbt removeTag "texture" else nbt setString("texture", s)
//  }
//
//  def getColourIndex(nbt: NBTTagCompound): Int = {
//    if (nbt hasKey "colourindex") nbt getInteger "colourindex" else defaultcolourindex
//  }
//
//  def setColourIndex(nbt: NBTTagCompound, c: Int) {
//    if (c == defaultcolourindex) nbt removeTag "colourindex" else nbt setInteger("colourindex", c)
//  }
//
//  def getGlow(nbt: NBTTagCompound): Boolean = {
//    if (nbt hasKey "glow") nbt getBoolean "glow" else defaultglow
//  }
//
//  def setGlow(nbt: NBTTagCompound, g: Boolean) {
//    if (g == defaultglow) nbt removeTag "glow" else nbt setBoolean("glow", g)
//  }
//
//  def setModel(nbt: NBTTagCompound, model: ModelSpec) {
//    setModel(nbt, ModelRegistry.inverse.getOrElse(model, ""))
//  }
//
//  def setModel(nbt: NBTTagCompound, modelname: String) {
//    nbt setString("model", modelname)
//  }
//
//  def setPart(nbt: NBTTagCompound) {
//    nbt setString("part", partName)
//  }
//
//  def setPartAndModel(nbt: NBTTagCompound) {
//  }
//
//  def multiSet(nbt: NBTTagCompound, tex: Option[String], glow: Option[Boolean], c: Option[Int]): NBTTagCompound = {
//    setPart(nbt)
//    setModel(nbt, this.modelSpec)
//    setTexture(nbt, tex.getOrElse(""))
//    setGlow(nbt, glow.getOrElse(false))
//    setColourIndex(nbt, c.getOrElse(defaultcolourindex))
//    nbt
//  }
//
//}

File: src/main/java/net/machinemuse/powersuits/client/render/modelspec/ModelPartSpec.java
package net.machinemuse.powersuits.client.render.modelspec;

import net.minecraft.client.renderer.block.model.BakedQuad;
import net.minecraft.client.resources.I18n;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import java.util.List;
import java.util.Objects;

/**
 * Ported to Java by lehjr on 11/8/16.
 */
@SideOnly(Side.CLIENT)
public class ModelPartSpec extends PartSpecBase {
    private final boolean defaultglow;

    public ModelPartSpec(final ModelSpec modelSpec,
                         final SpecBinding binding,
                         final String partName,
                         final Integer defaultcolourindex,
                         final Boolean defaultglow) {
        super(modelSpec, binding, partName, defaultcolourindex);
        this.defaultglow = (defaultglow != null) ? defaultglow : false;
    }

    @Override
    public String getDisaplayName() {
        return I18n.format(new StringBuilder("model.")
                .append(this.spec.getOwnName())
                .append(".")
                .append(this.partName)
                .append(".partName")
                .toString());
    }

    public boolean getGlow() {
        return this.defaultglow;
    }

    public boolean getGlow(NBTTagCompound nbt) {
        return nbt.hasKey("glow") ? nbt.getBoolean("glow") : this.defaultglow;
    }

    public void setGlow(NBTTagCompound nbt, boolean g) {
        if (g == this.defaultglow) nbt.removeTag("glow");
        else nbt.setBoolean("glow", g);
    }

    public List<BakedQuad> getQuads() {
        return ((ModelSpec) (this.spec)).getModel().getQuadsforPart(this.partName);
    }

    public NBTTagCompound multiSet(NBTTagCompound nbt, Integer colourIndex, Boolean glow) {
        super.multiSet(nbt, colourIndex);
        this.setGlow(nbt, (glow != null) ? glow : false);
        return nbt;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        ModelPartSpec that = (ModelPartSpec) o;
        return defaultglow == that.defaultglow;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), defaultglow);
    }
}

